definitions/src/instructions.rs

```diff
@@ -24,141 +24,250 @@ pub type InstructionOpcode = u8;
-pub const OP_CUSTOM_LOAD_IMM: InstructionOpcode = 103;
-pub const OP_ADDI: InstructionOpcode = 104;
-pub const OP_CUSTOM_TRACE_END: InstructionOpcode = 105;
+pub const OP_ADDI: InstructionOpcode = 2;
+pub const OP_ADDIW: InstructionOpcode = 3;
+pub const OP_ADDW: InstructionOpcode = 4;
+pub const OP_AND: InstructionOpcode = 5;
+pub const OP_ANDI: InstructionOpcode = 6;
+pub const OP_AUIPC: InstructionOpcode = 7;
+pub const OP_BEQ: InstructionOpcode = 8;
+pub const OP_BGE: InstructionOpcode = 9;
+pub const OP_BGEU: InstructionOpcode = 10;
+pub const OP_BLT: InstructionOpcode = 11;
+pub const OP_BLTU: InstructionOpcode = 12;
+pub const OP_BNE: InstructionOpcode = 13;
+pub const OP_DIV: InstructionOpcode = 14;
+pub const OP_DIVU: InstructionOpcode = 15;
+pub const OP_DIVUW: InstructionOpcode = 16;
+pub const OP_DIVW: InstructionOpcode = 17;
+pub const OP_EBREAK: InstructionOpcode = 18;
+pub const OP_ECALL: InstructionOpcode = 19;
+pub const OP_FENCE: InstructionOpcode = 20;
+pub const OP_FENCEI: InstructionOpcode = 21;
+pub const OP_JAL: InstructionOpcode = 22;
+pub const OP_JALR: InstructionOpcode = 23;
+pub const OP_LB: InstructionOpcode = 24;
+pub const OP_LBU: InstructionOpcode = 25;
+pub const OP_LD: InstructionOpcode = 26;
+pub const OP_LH: InstructionOpcode = 27;
+pub const OP_LHU: InstructionOpcode = 28;
+pub const OP_LUI: InstructionOpcode = 29;
+pub const OP_LW: InstructionOpcode = 30;
+pub const OP_LWU: InstructionOpcode = 31;
+pub const OP_MUL: InstructionOpcode = 32;
+pub const OP_MULH: InstructionOpcode = 33;
+pub const OP_MULHSU: InstructionOpcode = 34;
+pub const OP_MULHU: InstructionOpcode = 35;
+pub const OP_MULW: InstructionOpcode = 36;
+pub const OP_OR: InstructionOpcode = 37;
+pub const OP_ORI: InstructionOpcode = 38;
+pub const OP_REM: InstructionOpcode = 39;
+pub const OP_REMU: InstructionOpcode = 40;
+pub const OP_REMUW: InstructionOpcode = 41;
+pub const OP_REMW: InstructionOpcode = 42;
+pub const OP_SB: InstructionOpcode = 43;
+pub const OP_SD: InstructionOpcode = 44;
+pub const OP_SH: InstructionOpcode = 45;
+pub const OP_SLL: InstructionOpcode = 46;
+pub const OP_SLLI: InstructionOpcode = 47;
+pub const OP_SLLIW: InstructionOpcode = 48;
+pub const OP_SLLW: InstructionOpcode = 49;
+pub const OP_SLT: InstructionOpcode = 50;
+pub const OP_SLTI: InstructionOpcode = 51;
+pub const OP_SLTIU: InstructionOpcode = 52;
+pub const OP_SLTU: InstructionOpcode = 53;
+pub const OP_SRA: InstructionOpcode = 54;
+pub const OP_SRAI: InstructionOpcode = 55;
+pub const OP_SRAIW: InstructionOpcode = 56;
+pub const OP_SRAW: InstructionOpcode = 57;
+pub const OP_SRL: InstructionOpcode = 58;
+pub const OP_SRLI: InstructionOpcode = 59;
+pub const OP_SRLIW: InstructionOpcode = 60;
+pub const OP_SRLW: InstructionOpcode = 61;
+pub const OP_SUB: InstructionOpcode = 62;
+pub const OP_SUBW: InstructionOpcode = 63;
+pub const OP_SW: InstructionOpcode = 64;
+pub const OP_VERSION1_JALR: InstructionOpcode = 65;
```

Comment:
Any reason to allocate different instructon opcode? From the code base, it seems that `OP_VERSION1_JALR` and `OP_JALR` lead to identical logic.

Reply:
The 2 opcodes actually do use different logic:

https://github.com/nervosnetwork/ckb-vm/blob/01cf746f32af6be7be2b574b14fe660e63df4719/src/machine/asm/execute.S#L448-L456
https://github.com/nervosnetwork/ckb-vm/blob/01cf746f32af6be7be2b574b14fe660e63df4719/src/machine/asm/execute.S#L1093-L1101

Notice the first 4 lines in each version, they are in different orders. The idea is that version 0 can use the original `OP_JALR`, while version 1 and later uses `OP_VERSION1_JALR`